From 73786f30ed287435c32a87860abfc1d85a523f6f Mon Sep 17 00:00:00 2001
From: Xiao He <xiao.he@intel.com>
Date: Tue, 15 Dec 2020 16:44:38 +0800
Subject: [PATCH] vm-manager:Add BLKDISCARD & BLKDISCARDZEROES ioctl support

This option will enable the above ioctl support
in qemu. Android factory reset service need to
use BLKDISCARD to erase persistent partition.

Tracked-On: OAM-OAM-95481
Signed-off-by: Xiao He <xiao.he@intel.com>
---
 scripts/start_civ.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/start_civ.sh b/scripts/start_civ.sh
index 4e27b70..c7f7f14 100755
--- a/scripts/start_civ.sh
+++ b/scripts/start_civ.sh
@@ -229,7 +229,7 @@ function set_cpu() {
 }

 function set_disk() {
-    GUEST_DISK="-drive file=$1,if=none,id=disk1"
+    GUEST_DISK="-drive file=$1,if=none,id=disk1,discard=unmap,detect-zeroes=unmap"
     if [[ $1 =~ "/dev/" ]]; then
         SECURE_ERASE_DAEMON=$SCRIPTS_DIR/secure_erase_daemon
         local start=$(gdisk -l $1 | while read line;do vars=($line); [ "${vars[6]}" == "userdata" ] && echo ${vars[1]//.}; done)
--
2.17.1

